app/src/main/kotlin/com/wire/android/ui/home/conversations/details/AddMembersToConversationViewModel.kt

```diff
@@ -83,35 +86,35 @@ class AddMembersToConversationViewModel @Inject constructor(
             }
         }
 
-    override suspend fun searchKnownPeople(searchTerm: String): ContactSearchResult.InternalContact {
-        val result = withContext(dispatchers.io()) {
-            searchKnownUsers(
-                searchQuery = searchTerm,
-                searchUsersOptions = SearchUsersOptions(
-                    conversationExcluded = ConversationMemberExcludedOptions.ConversationExcluded(conversationId),
-                    selfUserIncluded = false
-                )
-            )
-        }
-
-        return when (result) {
-            is KnownUserSearchResult.Failure.Generic -> ContactSearchResult.InternalContact(
-                SearchResultState.Failure(R.string.label_general_error)
+    override suspend fun searchKnownPeople(searchTerm: String): Flow<ContactSearchResult.InternalContact> {
+       return searchKnownUsers(
+            searchQuery = searchTerm,
+            searchUsersOptions = SearchUsersOptions(
+                conversationExcluded = ConversationMemberExcludedOptions.ConversationExcluded(conversationId),
+                selfUserIncluded = false
             )
-            KnownUserSearchResult.Failure.InvalidQuery -> ContactSearchResult.InternalContact(
-                SearchResultState.Failure(R.string.label_general_error)
-            )
-            KnownUserSearchResult.Failure.InvalidRequest -> ContactSearchResult.InternalContact(
-                SearchResultState.Failure(R.string.label_general_error)
-            )
-            is KnownUserSearchResult.Success -> ContactSearchResult.InternalContact(
-                SearchResultState.Success(
-                    result.userSearchResult.result.map(
-                        contactMapper::fromOtherUser
+        )
+            .flowOn(dispatchers.io())
+            .map { result ->
+                when (result) {
+                    is KnownUserSearchResult.Failure.Generic -> ContactSearchResult.InternalContact(
+                        SearchResultState.Failure(R.string.label_general_error)
                     )
-                )
-            )
-        }
+                    KnownUserSearchResult.Failure.InvalidQuery -> ContactSearchResult.InternalContact(
+                        SearchResultState.Failure(R.string.label_general_error)
+                    )
+                    KnownUserSearchResult.Failure.InvalidRequest -> ContactSearchResult.InternalContact(
+                        SearchResultState.Failure(R.string.label_general_error)
+                    )
+                    is KnownUserSearchResult.Success -> ContactSearchResult.InternalContact(
+                        SearchResultState.Success(
+                            result.userSearchResult.result.map(
+                                contactMapper::fromOtherUser
+                            )
+                        )
+                    )
+                }
```

Comment:
Maybe move to a mapping method that can be unit tested?

Reply:
What should we test here? Id of strings? I just changed one id to be the same as on the designs 😅